Sources/SpeziHealthKit/CollectSample/CollectSample.swift

```diff
@@ -38,39 +41,55 @@ import Spezi
 ///
 /// ```swift
 /// CollectSample(
-///     HKQuantityType(.stepCount),
+///     .stepCount,
 ///     predicate: predicateOneMonth,
 ///     deliverySetting: .background(.automatic)
 /// )
 /// ```
-public struct CollectSample: HealthKitDataSourceDescription {
-    private let collectSamples: CollectSamples
+public struct CollectSample: HealthKitConfigurationComponent {
+    private let sampleType: HKSampleType
+    private let predicate: NSPredicate?
+    private let deliverySetting: HealthKitDeliverySetting
     
-    
-    public var sampleTypes: Set<HKSampleType> {
-        collectSamples.sampleTypes
+    public var dataAccessRequirements: HealthKitDataAccessRequirements {
+        .init(read: [sampleType])
     }
     
     
     /// - Parameters:
-    ///   - sampleType: The `HKSampleType` that should be collected
+    ///   - sampleType: The ``HealthKitSampleType`` that should be collected
     ///   - predicate: A custom predicate that should be passed to the HealthKit query.
     ///                The default predicate collects all samples that have been collected from the first time that the user
     ///                provided the application authorization to collect the samples.
     ///   - deliverySetting: The ``HealthKitDeliverySetting`` that should be used to collect the sample type. `.manual` is the default argument used.
-    public init<S: HKSampleType>(
-        _ sampleType: S,
+    public init(
+        _ sampleType: HealthKitSampleType<some Any>,
+        predicate: NSPredicate? = nil,
+        delivery: HealthKitDeliverySetting = .manual() // TODO Question @Paul : why does it default to manual?
```

Comment:
Because we didn't really have a good read/write setup before. I would argue that anchorQuery might make more sense now that we have a better mechanism to request dedicated read and write access for samples.

Reply:
@PSchmiedmayer now that the auth request handling is centralized, and multiple `CollectSample` configs trying to start automatically at the same time works properly; should we change the default here to `.automatic`?